### PR TITLE
chore: enable CodeQL scanning

### LINKED ISSUE #41: Add CodeQL
Enable CodeQL by default

### PR DESCRIPTION
**Pull request by gbartolini**
Closes #41

### CODE REVIEW

**Review by sxd — APPROVED** [review]

File: pkg/management/barman/archiver/archiver.go [hunk 1]
```diff
@@ -148,7 +149,12 @@ func (archiver *WALArchiver) ArchiveList(
 // Archive archives a certain WAL file using barman-cloud-wal-archive.
 // See archiveWALFileList for the meaning of the parameters
 func (archiver *WALArchiver) Archive(walName string, baseOptions []string) error {
-	options := make([]string, len(baseOptions), len(baseOptions)+1)
+	optionsLength := len(baseOptions)
+	optionsLengthMax := optionsLength + 1
+	if optionsLength >= math.MaxInt || optionsLengthMax >= math.MaxInt {
```
Comment (jsilvela) [inline pkg/management/barman/archiver/archiver.go:154]: this condition is redundant. It is enough to check `optionsLengthMax >= math.MaxInt`. Unless the codeql thing complains about that....

File: pkg/management/barman/restorer/restorer.go [hunk 2]
```diff
@@ -219,7 +220,12 @@ func (restorer *WALRestorer) RestoreList(
 
 // Restore restores a WAL file from the object store
 func (restorer *WALRestorer) Restore(walName, destinationPath string, baseOptions []string) error {
-	options := make([]string, len(baseOptions), len(baseOptions)+2)
+	optionsLength := len(baseOptions)
+	optionsLengthMax := optionsLength + 2
+	if optionsLength >= math.MaxInt || optionsLengthMax >= math.MaxInt {
```
Comment (jsilvela) [inline pkg/management/barman/restorer/restorer.go:225]: another redundant condition. `optionsLengthMax >= math.MaxInt` is enough ... unless codeql can't figure that out...

File: pkg/utils/barman.go [hunk 3]
```diff
@@ -42,11 +42,12 @@ func ParsePolicy(policy string) (string, error) {
...
 // Barman tags format needed
 func MapToBarmanTagsFormat(option string, mapTags map[string]string) []string {
-	if len(mapTags) == 0 {
+	tagsLength := len(mapTags)
+	if tagsLength == 0 {
 		return []string{}
 	}
 
-	tags := make([]string, 0, len(mapTags)+1)
+	tags := make([]string, 0, tagsLength)
```
Comment (jsilvela) [inline pkg/utils/barman.go:50]: since we're adding the option to the tags, we're short by 1 ... which we had in the original...

**Review by jsilvela — COMMENTED** [review]
Nice work! Two nits.
And one allocation size issue I'd like to clarify.

**Review by jsilvela — APPROVED** [review]
We could think of creating a function `func CheckSizeIncrement(initial, increment int) (int, error)` or similar to detect overflows.
But is OK as is.

**Review by phisco — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Review by armru — APPROVED** [review]

**Pull request merged**